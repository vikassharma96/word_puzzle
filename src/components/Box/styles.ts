import {StyleSheet} from 'react-native';
import colors from '../../styles/colors';

const styles = StyleSheet.create({
  box: {
    width: 40,
    height: 40,
    backgroundColor: colors.light.lightGray,
    margin: '4%',
    alignItems: 'center',
    justifyContent: 'center',
  },
  selectedBox: {
    backgroundColor: '#aaa',
  },
  letter: {
    textAlign: 'center',
  },
});

export default styles;
