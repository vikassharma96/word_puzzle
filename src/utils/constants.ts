// Home screen constants
export const WORDS_PUZZLE = 'Words Puzzle';
export const API_ERROR_MSG = "Couldn't retrieve the categories.";
export const START = 'Start';
export const LEADERS_BOARD = 'Leaders Board';
export const ERROR_MSG = 'Please select category first';

// Puzzle screen constants
export const WRONG_TEXT = 'Sorry! \n Bad luck';
export const CORRECT_TEXT = 'Correct! \n Congratulations';
export const EARN_TEXT = 'You earn';
export const POINTS = 'Points';
export const SCORE_SUMMARY = 'Score Summay';
export const TOTAL_SCORE = 'Your total score is';
export const SHARE = 'Share';

// common constants
export const ERROR = 'Error!';
export const RETRY = 'Retry';
export const BUTTON_TEXT_OK = 'Ok';
export const BUTTON_TEXT_CANCEL = 'Cancel';
export const SKIP = 'Skip';
export const NEXT = 'Next';
export const NO_DATA_AVAILABLE = 'No data available!';
export const WORD_PUZZLE_TEXT = 'Word Puzzle!';
export const SOMETHING_WENT_WRONG = 'Something went wrong!';
export const EXIT = 'Exit';
