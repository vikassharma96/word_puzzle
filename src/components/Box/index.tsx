import React, {useState} from 'react';
import {TouchableHighlight} from 'react-native';
import AppText from '../AppText';
import styles from './styles';

type IBoxProps = {
  text: string;
};

const Box = ({text}: IBoxProps) => {
  const [isSelected, setIsSelected] = useState<boolean>(false);
  return (
    <TouchableHighlight
      testID="box"
      style={[styles.box, isSelected && styles.selectedBox]}
      onPress={() => {
        if (!isSelected) {
          setIsSelected(true);
        }
      }}>
      <AppText style={styles.letter}>{!isSelected && text}</AppText>
    </TouchableHighlight>
  );
};

export default Box;
