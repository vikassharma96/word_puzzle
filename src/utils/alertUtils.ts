import {Alert} from 'react-native';
import {BUTTON_TEXT_CANCEL, BUTTON_TEXT_OK} from './constants';

type AlertFunction = {
  title: string;
  message: string;
  singleButton?: boolean;
  positiveText?: string;
  negativeText?: string;
  cancelable?: boolean;
  positiveAction?: () => void;
  negativeAction?: () => void;
  positiveActionStyle?: any;
  negativeActionStyle?: any;
};

export const showAlert = ({
  title,
  message,
  singleButton = true,
  positiveText = BUTTON_TEXT_OK,
  negativeText = BUTTON_TEXT_CANCEL,
  cancelable = false,
  positiveAction,
  negativeAction,
  positiveActionStyle,
  negativeActionStyle,
}: AlertFunction) => {
  const actionButtons = [
    {text: positiveText, onPress: positiveAction, style: positiveActionStyle},
  ];
  if (!singleButton) {
    actionButtons.unshift({
      text: negativeText,
      onPress: negativeAction,
      style: negativeActionStyle,
    });
  }
  Alert.alert(title, message, actionButtons, {
    cancelable: cancelable,
  });
};
