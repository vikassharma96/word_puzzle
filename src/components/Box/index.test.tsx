import React from 'react';
import renderer from 'react-test-renderer';
import Box from '.';
import {WORD_PUZZLE_TEXT} from '../../utils/constants';

describe('@components/Box', () => {
  const component = <Box text={WORD_PUZZLE_TEXT} />;

  it('should match snapshots', () => {
    const componentTree = renderer.create(component).toJSON();
    expect(componentTree).toMatchSnapshot();
  });
});
