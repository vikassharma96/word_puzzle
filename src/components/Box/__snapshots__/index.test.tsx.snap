// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/Box should match snapshots 1`] = `
<View
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#ccc",
        "height": 40,
        "justifyContent": "center",
        "margin": "4%",
        "width": 40,
      },
      false,
    ]
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Poppins-Medium",
          "fontSize": 14,
        },
        {
          "color": "#000",
        },
        {
          "textAlign": "center",
        },
      ]
    }
  >
    Word Puzzle!
  </Text>
</View>
`;
